R/Connection.R

```diff
@@ -494,45 +494,57 @@ setMethod(
   "dbQuoteIdentifier", c("OdbcConnection", "SQL"),
   getMethod("dbQuoteIdentifier", c("DBIConnection", "SQL"), asNamespace("DBI")))
 
-#' @inherit DBI::dbListTables
-#' @param catalog_name The name of the catalog to return, the default returns all catalogs.
-#' @param schema_name The name of the schema to return, the default returns all schemas.
-#' @param table_name The name of the table to return, the default returns all tables.
+#' List remote tables and fields for an ODBC connection
+#'
+#' `dbListTables()` provides names of remote tables accessible through this
+#' connection; `dbListFields()` provides names of columns within a table.
+#'
+#' @inheritParams DBI::dbListTables
+#' @param catalog_name,schema_name,table_name Catalog, schema, and table names.
+#'
+#'   Use `%` as a wildcard to match any name; use `_` to any any character.
 #' @param table_type The type of the table to return, the default returns all table types.
-#' @aliases dbListTables
```

Comment:
Noting that this change, and the analogous one for the `dbListFields` method, means that with only odbc loaded (i.e. not having called `library(DBI)`) `?dbListTables` will now head to the generic's docs in DBI rather than the "pick the place where you want to go to" docs:

<img width="693" alt="Screenshot 2023-12-07 at 2 28 13 PM" src="https://github.com/r-dbi/odbc/assets/35748691/5f71526b-6a57-43ba-a7b6-60991f0f0fd4">

This sounds good to me, since DBI's docs link to the odbc method's docs early on, but is this a change we want to make generally?